langchain/src/chat_models/universal.ts

```diff
@@ -0,0 +1,870 @@
+      Object.entries(mergedConfig).filter(([k]) => k !== "configurable")
+    );
+
+    remainingConfig.configurable = Object.fromEntries(
+      Object.entries(mergedConfig.configurable || {}).filter(
+        ([k]) =>
+          this._configPrefix &&
+          !Object.keys(modelParams).includes(
+            this._removePrefix(k, this._configPrefix)
+          )
+      )
+    );
+
+    const newConfigurableModel = new _ConfigurableModel<RunInput, CallOptions>({
+      defaultConfig: { ...this._defaultConfig, ...modelParams },
+      configurableFields: Array.isArray(this._configurableFields)
+        ? [...this._configurableFields]
+        : this._configurableFields,
+      configPrefix: this._configPrefix,
+    });
+
+    return new RunnableBinding<RunInput, AIMessageChunk, CallOptions>({
+      config: mergedConfig,
+      bound: newConfigurableModel,
+    });
+  }
+
+  async invoke(
+    input: RunInput,
+    options?: CallOptions
+  ): Promise<AIMessageChunk> {
+    const model = await this._model(options);
+    const config = ensureConfig(options);
+    return model.invoke(input, config);
+  }
+
+  async stream(
+    input: RunInput,
+    options?: CallOptions
+  ): Promise<IterableReadableStream<AIMessageChunk>> {
+    const model = await this._model(options);
+    const wrappedGenerator = new AsyncGeneratorWithSetup({
+      generator: await model.stream(input, options),
+      config: options,
+    });
+    await wrappedGenerator.setup;
+    return IterableReadableStream.fromAsyncGenerator(wrappedGenerator);
+  }
+
+  async batch(
+    inputs: RunInput[],
+    options?: Partial<CallOptions> | Partial<CallOptions>[],
+    batchOptions?: RunnableBatchOptions & { returnExceptions?: false }
+  ): Promise<AIMessageChunk[]>;
+
+  async batch(
+    inputs: RunInput[],
+    options?: Partial<CallOptions> | Partial<CallOptions>[],
+    batchOptions?: RunnableBatchOptions & { returnExceptions: true }
+  ): Promise<(AIMessageChunk | Error)[]>;
+
+  async batch(
+    inputs: RunInput[],
+    options?: Partial<CallOptions> | Partial<CallOptions>[],
+    batchOptions?: RunnableBatchOptions
+  ): Promise<(AIMessageChunk | Error)[]>;
+
+  async batch(
+    inputs: RunInput[],
+    options?: Partial<CallOptions> | Partial<CallOptions>[],
+    batchOptions?: RunnableBatchOptions
+  ): Promise<(AIMessageChunk | Error)[]> {
+    // If options is undefined, null, an object, or an array with 0 or 1 element
+    if (
+      !options ||
+      !Array.isArray(options) ||
+      (Array.isArray(options) && options.length <= 1)
+    ) {
+      const config = Array.isArray(options) ? options[0] : options;
+      const model = await this._model(config);
+      return model.batch(inputs, config, {
+        ...batchOptions,
+        returnExceptions: batchOptions?.returnExceptions ?? false,
+      });
+    } else {
+      // If multiple configs, use the base Runnable.batch implementation
+      return super.batch(inputs, options, batchOptions);
```

Comment:
Why?

Reply:
Generally would prefer to preserve base model behavior

Reply:
Refactored to just call `super.batch` and not implement any logic ourselves. Left over from PY port (not sure why they do it)